loky/backend/utils.py

```diff
@@ -13,17 +17,41 @@ def _flag_current_thread_clean_exit():
     thread._clean_exit = True
 
 
-def recursive_terminate(process):
+def recursive_terminate(process, use_psutil=True):
+    if use_psutil and psutil is not None:
+        _recursive_terminate_with_psutil(process)
+    else:
+        _recursive_terminate_without_psutil(process)
+
+
+def _recursive_terminate_with_psutil(process, retries=5):
+    try:
+        children = psutil.Process(process.pid).children(recursive=True)
+    except psutil.NoSuchProcess:
+        return
+
+    for child in children:
+        try:
+            child.terminate()
```

Comment:
We should directly call `kill` as the goal of `recursive_terminate` is to forcefully clean-up the remaining `worker` when the `executor` is not in a proper state.

Right now, this makes it very slow to kill workers and the test `max_depth` takes more than 1:30min. Another solution is to reduce the timeout line 39.